File: frontend/src/services/FlightService.js
import axios from 'axios';


// Create an instance of axios with the base URL
const flightServiceInstance = axios.create({
    baseURL: FLIGHT_API_BASE_URL,
});

class FlightService {
    getAllFlights() {
        return flightServiceInstance.get('/');
    }

    getFlightById(id) {
        return flightServiceInstance.get(`/${id}`);
    }

    createFlight(flight) {
        return flightServiceInstance.post('/', flight);
    }

    updateFlight(id, flight) {
        return flightServiceInstance.put(`/${id}`, flight);
    }

    deleteFlight(id) {
        return flightServiceInstance.delete(`/${id}`);
    }
}

export default new FlightService();

File: src/main/java/Airlines/AirlineTicket/Flight/Models/FlightService.java
package Airlines.AirlineTicket.Flight.Models;

import Airlines.AirlineTicket.Flight.Flight;

import java.util.List;

public interface FlightService {

    List<Flight> getAllFlights();

    Flight getFlightById(Long id);

    Flight saveFlight(Flight flight);

    void deleteFlight(Long id);
}

